lib/collection/src/update_handler.rs

```diff
@@ -257,6 +260,22 @@ impl UpdateHandler {
     {
         let mut scheduled_segment_ids = HashSet::<_>::default();
         let mut handles = vec![];
+
+        // Ensure we have at least one appendable segment with enough capacity
+        // Source required parameters from first optimizer
+        if let Some(optimizer) = optimizers.first() {
+            let result = Self::ensure_appendable_segment_with_capacity(
+                &segments,
+                optimizer.segments_path(),
+                &optimizer.collection_params(),
+                optimizer.threshold_config(),
+            );
+            if let Err(err) = result {
+                log::error!("Failed to ensure there are appendable segments with capacity: {err}");
+                panic!("Failed to ensure there are appendable segments with capacity: {err}");
+            }
+        }
```

Comment:
Here - in the optimization loop - we now ensure we always have at least one segment with capacity. We check this before running any of the optimizers and their conditions.